api/src/testIntegration/java/bisq/api/http/UserEndpointIT.java

```diff
@@ -0,0 +1,207 @@
+        verifyThatPasswordIsValid(validPassword);
+        verifyThatPasswordIsInvalid(newPassword);
+    }
+
+    @InSequence(4)
+    @Test
+    public void changePassword_emptyOldPassword_returns401() {
+        String newPassword = getRandomPasswordDifferentThan(validPassword);
+        given().
+                port(getAlicePort()).
+                body(new ChangePassword(newPassword, null)).
+                contentType(ContentType.JSON).
+//
+        when().
+                post("/api/v1/user/password").
+//
+        then().
+                statusCode(401);
+        verifyThatAuthenticationIsEnabled();
+        verifyThatPasswordIsValid(validPassword);
+        verifyThatPasswordIsInvalid(newPassword);
+    }
+
+
+    @InSequence(5)
+    @Test
+    public void changePassword_settingAnotherPassword_keepsAuthenticationEnabled() {
+        String oldPassword = validPassword;
+        String newPassword = getRandomPasswordDifferentThan(validPassword);
+        validPassword = newPassword;
+        invalidPassword = getRandomPasswordDifferentThan(validPassword);
+        given().
+                port(getAlicePort()).
+                body(new ChangePassword(newPassword, oldPassword)).
+                contentType(ContentType.JSON).
+//
+        when().
+                post("/api/v1/user/password").
+//
+        then().
+                statusCode(204);
+        verifyThatPasswordIsInvalid(oldPassword);
+        verifyThatPasswordIsValid(newPassword);
+        verifyThatAuthenticationIsEnabled();
+    }
+
+    @InSequence(6)
+    @Test
+    public void changePassword_validOldPasswordAndNoNewPassword_disablesAuthentication() {
+        given().
+                port(getAlicePort()).
+                body(new ChangePassword(null, validPassword)).
+                contentType(ContentType.JSON).
+//
+        when().
+                post("/api/v1/user/password").
+//
+        then().
+                statusCode(204);
+        verifyThatAuthenticationIsDisabled();
+    }
+
+    private void verifyThatAuthenticationIsDisabled() {
+        authenticationVerificationRequest().then().statusCode(200);
+    }
+
+    private void verifyThatAuthenticationIsEnabled() {
+        authenticationVerificationRequest().then().statusCode(401);
+    }
+
+    private void verifyThatPasswordIsInvalid(String password) {
+        passwordVerificationRequest(password).then().statusCode(401);
+    }
+
+    private void verifyThatPasswordIsValid(String password) {
+        passwordVerificationRequest(password).then().statusCode(200);
+    }
+
+    private Response authenticationVerificationRequest() {
+        return given().port(getAlicePort()).when().get("/api/v1/version");
+    }
+
+    private Response passwordVerificationRequest(String password) {
+        String authHeader = "Basic " + Base64.encode((":" + password).getBytes());
+        return given().port(getAlicePort()).when()
+                .header("authorization", authHeader)
```

Comment:
`authHeader` also seems to be redundant here - we can use 

    .header("authorization", "Basic " + Base64.encode((":" + password).getBytes()))

I see also different styling here, I mean `given().port(...).when()` are in single line. It's private method, but I think we can unify it.

Reply:
Inlining authHeader makes it harder to read.
The variable with good name gives you immediate idea what the code is doing.

Reply:
I will not press but line like:

    .header("authorization", "Basic " + Base64.encode((":" + password).getBytes()))

is readable enough. I immediately see that `"Basic " + Base64.encode((":" + password).getBytes()` is value for authorization header, and `authHeader` variable name has no value here. 

It's minor thing, so it's ok for me to left `authHeader` as it is.